test/k8sT/Policies.go

```diff
@@ -1403,6 +1406,29 @@ var _ = Describe("K8sPolicyTest", func() {
 				})
 			})
 		})
+
+		Context("with L7 policy", func() {
+			BeforeAll(func() {
+				if helpers.RunsOnNetNextKernel() {
+					By("Reconfiguring Cilium to enable BPF TProxy")
+					RedeployCiliumWithMerge(kubectl, ciliumFilename, daemonCfg,
+						map[string]string{
+							"config.bpfTProxy": "true",
+						})
+				}
+			})
+
+			AfterEach(func() {
+				kubectl.Delete(connectivityCheckYml)
+			})
+
+			It("using connectivity-check to check datapath", func() {
+				kubectl.ApplyDefault(connectivityCheckYml).ExpectSuccess("cannot install connectivity-check")
+
+				err := kubectl.WaitforPods(helpers.DefaultNamespace, "", helpers.HelperTimeout)
+				ExpectWithOffset(1, err).Should(BeNil(), "connectivity-check pods are not ready after timeout")
```

Comment:
`ExpectWithOffset` will show that the test failed at the *caller* of this `It()` block (which would be somewhere in ginkgo-ext or similar), IIRC. I think what you want is just an `Expect`, because that'll generate a stacktrace when the test fails, pointing at this specific line, which when a user looks at, it is clear that the error was with `WaitforPods`.

In general, I think if you are doing an expectation inside the `It()` block, you should use a simple `Expect`. An example of when you would use `ExpectWithOffset` is when you've defined a function that *performs* expectations. That's when you'd want to "offset" it. Does that make sense?

Reply:
Good catch, I was copy/pasting this from a similar example elsewhere in the testsuite and didn't pay close attention. Will fix it up.